workflows/sarek/main.nf

```diff
@@ -253,6 +253,12 @@ workflow SAREK {
             }
             .set { reads_grouping_key }
 
+        reads_for_alignment = reads_for_alignment.map{ meta, reads ->
+            // Update meta.id to meta.sample no multiple lanes or splitted fastqs
+            if (meta.size * meta.num_lanes == 1) [ meta + [ id:meta.sample ], reads ]
+            else [ meta, reads ]
+        }
+
```

Comment:
can this not be moved up to the previous mapping statement?

Reply:
or is it important that the mapping for the id is not stored in reads_grouping_key?

Reply:
Not sure what you mean, but you're welcome to move stuff around.

Reply:
Never mind, the other is Adam's cool grouping logic. One more concern from my side, below there is the joining between the `reads_grouping_key` and `reads_alignment` meta's. Is this working for both splitting and unsplitting and across several lanes?

Reply:
No idea 😬

You mean these combine-statements:

https://github.com/asp8200/sarek/blob/8d1592c242fbb2bd2cc7da66377032d55afc7816/workflows/sarek/main.nf#L270

https://github.com/asp8200/sarek/blob/8d1592c242fbb2bd2cc7da66377032d55afc7816/workflows/sarek/main.nf#L288

 I just hope we have CI-tests covering all this intricate stuff 🤞

Reply:
How about we merge as is for now, and we'll take care of beautifying it when we refactor?